src/main/java/htsjdk/samtools/util/SnappyLoader.java

```diff
@@ -57,71 +49,47 @@ public SnappyLoader() {
-                snappyErrorClass = (Class<Error>)Class.forName("org.xerial.snappy.SnappyError");
-                inputStreamCtor = snappyInputStreamClass.getConstructor(InputStream.class);
-                outputStreamCtor = snappyOutputStreamClass.getConstructor(OutputStream.class, Integer.TYPE);
+                final OutputStream test = new SnappyOutputStream(new ByteArrayOutputStream(1000));
+                test.write("Hello World!".getBytes());
+                test.close();
+                tmpSnappyAvailable = true;
+                if (logSuccesses) System.err.println("Snappy successfully loaded.");
             }
-            catch (NoSuchMethodException e) { /* Do nothing. */ }
-            catch (ClassNotFoundException e) { /* Do nothing. */ }
-        }
-
-        this.SnappyInputStreamCtor = inputStreamCtor;
-        this.SnappyOutputStreamCtor = outputStreamCtor;
-
-        if (this.SnappyInputStreamCtor != null && this.SnappyOutputStreamCtor != null) {
-            // Don't try to call any Snappy code until classes have been found via reflection above.
-            boolean tmpSnappyAvailable;
-            try {
-                if (!LoadSnappy.load()) {
-                    if (verbose) System.err.println("Snappy dll failed to load.");
-                    tmpSnappyAvailable = false;
-                }
-                else {
-                    if (verbose) System.err.println("Snappy stream classes loaded.");
-                    tmpSnappyAvailable = true;
-                }
-            } catch (Error e) {
-                if (e.getClass().equals(snappyErrorClass)) {
-                    if (verbose) System.err.println("Snappy dll failed to load: " + e.getMessage());
-                    tmpSnappyAvailable = false;
-                } else {
-                    throw e;
-                }
+            catch (Throwable e) {
+                if (logFailures) System.err.println("Snappy native library failed to load: " + e.getMessage());
             }
             SnappyAvailable = tmpSnappyAvailable;
         }
-        else {
-            if (verbose) System.err.println("Snappy stream classes not loaded.");
-            SnappyAvailable = false;
-        }
     }
 
+    /** Returns true if Snappy is available, false otherwise. */
+    public boolean isSnappyAvailable() { return SnappyAvailable; }
+
     /** Wrap an InputStream in a SnappyInputStream. If Snappy is not available will throw an exception. */
     public InputStream wrapInputStream(final InputStream inputStream) {
-        try {
-            return SnappyInputStreamCtor.newInstance(inputStream);
-        } catch (Exception e) {
-            throw new SAMException("Error instantiating SnappyInputStream", e);
-        }
+        try { return new SnappyInputStream(inputStream); }
+        catch (Exception e) { throw new SAMException("Error instantiating SnappyInputStream", e); }
```

Comment:
I think this should probably check`isSnappyAvailable` and throw if it's false.  It looks like maybe it used to check it implicitly by getting a `NullPointerException` if the constructor wasn't initialized.   As this is now you could specify `snappy.disable` on the command line and then end up using snappy unintentionally if you didn't guard against it in your own code.